File: src/main/java/com/example/skyr/note/sync/NotesSync.java
package com.example.skyr.note.sync;

import java.time.Instant;
import java.util.UUID;

public record NotesSync(UUID syncId, Instant syncTime) {
    public static final String SYNC_ID = "sync_id";
    public static final String SYNC_TIME = "sync_time";
}

File: src/main/kotlin/com/example/skyr/note/sync/NotesSync.kt
package com.example.skyr.note.sync

import com.example.skyr.getInstant
import com.example.skyr.getUuid
import java.sql.ResultSet
import java.time.Instant
import java.util.*

class NotesSync(val syncId: UUID = UUID.randomUUID(), val syncTime: Instant) {

    companion object {
        const val SYNC_ID = "sync_id"
        const val SYNC_TIME = "sync_time"
    }
}

fun notesSync(resultSet: ResultSet): NotesSync {
    val syncId = resultSet.getUuid(NotesSync.SYNC_ID)
    val syncTime = resultSet.getInstant(NotesSync.SYNC_TIME)
    return NotesSync(syncId, syncTime)
}
